dev-test/gql-tag-operations/graphql/fragment-masking.ts

```diff
@@ -46,3 +47,19 @@ export function makeFragmentData<F extends DocumentTypeDecoration<any, any>, FT
 ): FragmentType<F> {
   return data as FragmentType<F>;
 }
+export function isFragmentReady<TQuery, TFrag>(
+  queryNode: DocumentTypeDecoration<TQuery, any>,
+  fragmentNode: TypedDocumentNode<TFrag>,
+  data: Record<string, any>
```

Comment:
Shouldn't this be `TQuery` instead?

Reply:
Is there any way, we could make this more type-safe?

Reply:
Hmm, yeah. We could have a union and use recursive types. Will add that

Reply:
It actually turned out to be a bit simpler